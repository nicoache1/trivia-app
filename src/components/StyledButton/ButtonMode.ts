export enum ButtonMode {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
