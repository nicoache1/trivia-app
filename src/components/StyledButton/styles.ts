import { StyleSheet } from 'react-native'

import { ITheme } from '../../styles/types'
import { ButtonMode } from './ButtonMode'

export const textStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    color: theme.colors.WHITE,
  },
  [ButtonMode.LIGHT]: {
    color: theme.colors.WHITE,
  },
})

export const buttonStyle = (theme: ITheme) => ({
  [ButtonMode.DARK]: {
    backgroundColor: theme.colors.PRIMARY,
  },
  [ButtonMode.LIGHT]: {
    backgroundColor: theme.colors.SECONDARY,
  },
})

export const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    borderRadius: 10,
    height: 56,
    justifyContent: 'center',
    paddingVertical: 10,
    width: '100%',
  },
  buttonText: {
    alignItems: 'center',
  },
})
