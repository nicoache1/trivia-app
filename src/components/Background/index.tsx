import React from 'react'
import { useWindowDimensions, View } from 'react-native'
import Svg, { Path } from 'react-native-svg'

import { styles } from './styles'

export const Background: React.FC<{}> = ({}) => {
  const { width } = useWindowDimensions()

  return (
    <View style={styles.vector}>
      <Svg
        width={width}
        height="812"
        viewBox="0 0 375 812"
        fill="none"
        preserveAspectRatio="xMinYMin slice">
        <Path
          d="M-757.5 612.5L-1210 243.5V948.5H459V264.5L112 673L-335.5 -76L-757.5 612.5Z"
          fill="#142664"
          fillOpacity="0.3"
        />
      </Svg>
    </View>
  )
}
