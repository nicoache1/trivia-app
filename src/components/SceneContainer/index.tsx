import * as React from 'react'
import { StatusBar, StatusBarProps, StyleProp, ViewStyle } from 'react-native'
import { Edge, SafeAreaView } from 'react-native-safe-area-context'
import { useTheme } from 'src/styles/Theme'

import { styles } from './styles'

interface LayoutProps {
  edges?: ReadonlyArray<Edge>
  style?: StyleProp<ViewStyle>
  children: React.ReactNode
  barStyle?: StatusBarProps['barStyle']
}

export const SceneContainer: React.FunctionComponent<LayoutProps> = ({
  edges = ['bottom'],
  style,
  children,
  barStyle = 'default',
}) => {
  const { Theme } = useTheme()

  return (
    <SafeAreaView
      style={[styles.container, { backgroundColor: Theme.colors.WHITE }, style]}
      edges={edges}>
      <StatusBar barStyle={barStyle} animated={true} />
      {children}
    </SafeAreaView>
  )
}
