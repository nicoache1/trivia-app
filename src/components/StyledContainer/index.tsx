import { concat } from 'lodash'
import React from 'react'
import { StyleProp, View, ViewProps, ViewStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { ColorPalette } from '../../styles/types'

type CustomViewProps = {
  color?: ColorPalette
  children?: React.ReactNode
} & ViewProps

export const StyledContainer: React.FC<CustomViewProps> = props => {
  const { Theme: ActualTheme } = useTheme()

  const { color, style, children } = props
  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.TRANSPARENT

  const contextStyle: StyleProp<ViewStyle> = {
    backgroundColor: themedColor,
    paddingHorizontal: ActualTheme.spacing.HORIZONTAL_SCREEN_PADDING,
    paddingVertical: 10,
  }

  const mergedStyle: StyleProp<ViewStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <View {...props} style={mergedStyle}>
      {children}
    </View>
  )
}
