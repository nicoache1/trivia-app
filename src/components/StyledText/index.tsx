import { concat } from 'lodash'
import React from 'react'
import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { AppTypography, ColorPalette } from '../../styles/types'

type TextCustomProps = {
  typography?: AppTypography
  color?: ColorPalette
  children: React.ReactNode
} & TextProps

export const StyledText: React.FC<TextCustomProps> = props => {
  const { Theme: ActualTheme } = useTheme()
  const {
    color = ColorPalette.BLACK,
    children,
    typography = AppTypography.P1_REGULAR,
    style,
  } = props

  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.BLACK

  const themedTypography = ActualTheme.typography[typography]

  const contextStyle: StyleProp<TextStyle> = {
    color: themedColor,
    ...themedTypography,
  }

  const mergedStyle: StyleProp<TextStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <Text {...props} style={mergedStyle} accessible={true}>
      {children}
    </Text>
  )
}
