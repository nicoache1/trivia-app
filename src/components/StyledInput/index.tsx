import React from 'react'
import { TextInputProps } from 'react-native'
import { HelperText, TextInput } from 'react-native-paper'

import { useTheme } from '../../styles/Theme'
import { StyledContainer } from '../StyledContainer'
import { styles } from './styles'

interface StyledInputProps extends TextInputProps {
  helperText?: string
  label?: string
  error?: boolean
}

export const StyledInput: React.FC<StyledInputProps> = ({
  placeholder,
  label,
  error = false,
  helperText,
  ...props
}) => {
  const { Theme } = useTheme()

  const borderColor = error ? Theme.colors.ERROR : Theme.colors.WHITE

  return (
    <StyledContainer>
      <TextInput
        {...props}
        label={label}
        style={[
          styles.input,
          {
            borderColor,
          },
        ]}
        activeOutlineColor={Theme.colors.WHITE}
        placeholderTextColor={Theme.colors.WHITE}
        placeholder={placeholder}
        selectionColor={Theme.colors.WHITE}
        theme={{
          colors: {
            accent: borderColor,
            disabled: borderColor,
            placeholder: borderColor,
            primary: borderColor,
            text: borderColor,
          },
        }}
        mode={'flat'}
      />
      {!!helperText && (
        <HelperText type={'error'} visible={error}>
          {helperText}
        </HelperText>
      )}
    </StyledContainer>
  )
}
