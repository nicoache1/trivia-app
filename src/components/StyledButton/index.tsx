import React from 'react'
import { PressableProps, StyleProp, ViewStyle } from 'react-native'
import { Pressable } from 'react-native'
import Animated from 'react-native-reanimated'

import { useTheme } from '../../styles/Theme'
import { AppTypography } from '../../styles/types'
import { StyledText } from '../StyledText'
import { ButtonMode } from './ButtonMode'
import { buttonStyle, styles, textStyle } from './styles'

interface Props extends PressableProps {
  label: string
  mode?: ButtonMode
  customStyles?: StyleProp<ViewStyle>
}

export const StyledButton: React.FC<Props> = ({
  label,
  mode = ButtonMode.LIGHT,
  onPress,
  customStyles,
}) => {
  const { Theme } = useTheme()
  return (
    <Pressable
      style={({ pressed }) => [styles.button, { opacity: pressed ? 0.5 : 1.0 }]}
      onPress={onPress}>
      <Animated.View
        style={[
          styles.button,
          customStyles !== undefined ? customStyles : buttonStyle(Theme)[mode],
        ]}>
        <StyledText
          style={[styles.buttonText, textStyle(Theme)[mode]]}
          typography={AppTypography.P1_BOLD}>
          {label}
        </StyledText>
      </Animated.View>
    </Pressable>
  )
}
