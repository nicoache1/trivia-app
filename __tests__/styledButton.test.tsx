import { render } from '@testing-library/react-native'
import React from 'react'
import { StyledButton } from 'src/components/StyledButton'

test('Expect StyledButton to be rendered correctly', async () => {
  const renderedSnapshot = render(<StyledButton label={'label'} />).toJSON()

  expect(renderedSnapshot).toMatchSnapshot()
})
