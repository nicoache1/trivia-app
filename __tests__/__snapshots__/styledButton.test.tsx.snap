// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect StyledButton to be rendered correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "height": 56,
        "justifyContent": "center",
        "paddingVertical": 10,
        "width": "100%",
      },
      Object {
        "opacity": 1,
      },
    ]
  }
>
  <View
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(108,92,231)",
        "borderRadius": 10,
        "height": 56,
        "justifyContent": "center",
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "700",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
      typography="P1_BOLD"
    >
      label
    </Text>
  </View>
</View>
`;
